src/Language/PureScript/Ide/Filter.hs

```diff
@@ -77,6 +79,15 @@ declarationFilter :: (IdeDeclaration -> Text -> Bool) -> Text -> [Module] -> [Mo
 declarationFilter predicate search =
   filterModuleDecls (flip predicate search)
 
+-- | Only keeps Identifiers in the given type declarations
+declarationTypeFilter :: [D.IdeDeclaration] -> Filter
+declarationTypeFilter [] = mkFilter identity
+declarationTypeFilter decls =
+    mkFilter $ filterModuleDecls filterDecls
+    where
+      filterDecls :: IdeDeclaration -> Bool
+      filterDecls decl = D.typeDeclarationForDeclaration decl `elem` decls
```

Comment:
@kRITZCREEK Unfortunately I can't see what you mean here. Because `mkFilter` takes a function `([Module] -> [Module])`, but in your case it would be `Map ModuleName `???

Also `filterModuleDecls` as a part of `declarationTypeFilter` is already in `Language.PureScript.Ide.Filter` and it is used by other functions. Why should it moved (or maybe duplicate) into `Language.PureScript.Ide.Filter.Declaration`?

Could you give more information on this, pls?

P.S. Everything else is clear and has been already fixed with latest commit 7c680bd ;)